src/pages/chainlink-functions/resources/add-functions-to-projects.md

```diff
@@ -0,0 +1,317 @@
+
+:::note[Request Access]
+Chainlink Functions is currently in a limited BETA.
+Apply [here](http://functions.chain.link/) to add your EVM account address to the Allow List.
+:::
+
+### Configure Hardhat
+
+If you already have a project, you can skip to the [Libraries and Dependencies](#get-the-libraries-and-contract-dependencies) section. If you don't already have a Hardhat project, start one with the following steps:
+
+1. Install [Node.js](https://nodejs.org/en/download/).
+1. Make a new project directory:
+
+   ```shell
+   mkdir new-functions-project && cd new-functions-project
+   ```
+
+1. Create a new Node.js project:
+
+   ```shell
+   npm init
+   ```
+
+1. Install Hardhat as a dev dependency:
+
+   ```shell
+   npm install --save-dev hardhat
+   ```
+
+1. Install the dependency packages:
+
+   ```shell
+   npm install @chainlink/contracts @openzeppelin/contracts @openzeppelin/contracts-upgradeable eth-crypto dotenv axios vm2 is-http-url prompt-sync
+   ```
+
+1. Create a new Hardhat project. For this example, create a JavaScript project and use the default configurations:
+
+   ```shell
+   npx hardhat
+   ```
+
+1. Remove the `Lock.sol` contract, which Hardhat creates by default. This contract is not necessary for this tutorial.
+
+   ```shell
+   rm ./contracts/Lock.sol
+   ```
+
+### Configure Hardhat
+
+Modify `hardhat.config.js` to include the `0.8.7` and `0.8.9` compiler version and supported networks. See the [Supported Networks](/chainlink-functions/supported-networks) page for a list of networks where Chainlink Functions DONs are available. This example uses Polygon Mumbai by default:
+
+<CodeSample src="samples/Functions/adding-functions-to-projects/hardhat.config.js" />
+
+If you use this configuration with `require('dotenv').config()` to get your RPC URLs and private keys, you must also create a `.env` file with these variables set. Add your RPC URLs with your API keys and the private key to the wallet you will use to fund transactions on these testnets.
+
+```shell
+touch .env && open .env
+```
+
+```text
+# RPC URLs
+MUMBAI_RPC_URL="https://polygon-mumbai.infura.io/v3/your_api_key"
+SEPOLIA_RPC_URL="https://sepolia.infura.io/v3/your_api_key"
+
+# Wallet private key
+PRIVATE_KEY="your_private_key"
+```
+
+Run `npx hardhat compile` to make sure Hardhat is configured correctly. You will see several warnings, but everything is correct if the Solidity files compile successfully.
+
+## Get the dependency contracts and scripts
+
+Get the required libraries from the [Chainlink Functions Starter Kit](https://github.com/smartcontractkit/functions-hardhat-starter-kit). You can clone the repo and get the folders manually, or use a single command to get a tarball through the GitHub API and extract the folders you need to the correct location in your project:
```

Comment:
what folder do they run this git command from? The root folder of their project?

Reply:
I added a clarifying step to change directories to project root. See commit https://github.com/smartcontractkit/documentation/pull/1171/commits/619081a542894423bfb45b5dda8c18d7b628c7a0